### PR TITLE
BUG_FIX: Use inav in plot spectra

### PR DESCRIPTION
**Pull request by dnjohnstone**
This provides a fix for #880 so that the Signal object is indexed using inav rather than direct indexing.

### CODE REVIEW

File: hyperspy/drawing/utils.py [hunk 1]
```diff
@@ -315,7 +315,7 @@ def _make_overlap_plot(spectra, ax, color="blue", line_style='-'):
         x_axis = spectrum.axes_manager.signal_axes[0]
         ax.plot(x_axis.axis, spectrum.data, color=color, ls=line_style)
     if len(spectra) > 1:
```
Comment (vidartf) [inline hyperspy/drawing/utils.py]: `if not isinstance(spectra, Signal) and len(spectra) > 1:` might be better?

Comment (vidartf) [inline hyperspy/drawing/utils.py]: Although I have to admit I don't understand why the xlabel is only set `if len(spectra) > 1`...

**Comment by francisco-dlp** [comment]
I think that @vidartf is right, this function, unlike `plot_images`, wasn't designed to take spectra directly from a multidimensional `Signal`. `plot_images` is newer and enjoys this and some other features that it'll be nice to port to `plot_spectra`.

**Comment by dnjohnstone** [comment]
That may be true - but it doesn't change the fact that until now it was possible to use the line hs.plot.plot_spectra(s.get_decomposition_factors()) to get an overlay or cascade plot of results - and this is used in the EDS tutorial notebooks.

That's now not possible because you have to slice rather than index the output of get_decomposition_factors().... anyone who like me had lifted this line from the tutorials and find it useful to use on a day to day basis will now find this broken.

File: hyperspy/drawing/utils.py (hunk 1 above)
Comment (francisco-dlp) [inline hyperspy/drawing/utils.py]: @vidartf, and I think that that's a bug.

What about:

```python

_set_spectrum_xlabel(spectra if isinstance(spectra, Signal) else spectra[-1], ax)

```

**Comment by dnjohnstone** [comment]
The line from @francisco-dlp seems to work, so I've used that. Not totally sure why coveralls is complaining as it's also saying it's unchanged, but otherwise I think this is ok.

**Comment by vidartf** [comment]
@dnjohnstone : Since the new code is one line shorter than the previous version, ad that line was covered by tests, the ratio between covered and uncovered lines has declined. Don't worry about it :)

**Pull request merged**